### PR TITLE
OCPBUGS-41492: show Samples tab when one or more exists

### PR DESCRIPTION
**Pull request by rhamilto**
After

If creating new and sample and snippet exist, both tabs appear; if neither sample not snippet exist, no tabs show:

https://github.com/user-attachments/assets/52d9ab48-8e76-4ce4-9e9f-f6722908dada

If editing existing and sample and snippet exist, only snippet tab appears:

https://github.com/user-attachments/assets/24104507-43a0-446e-b694-bdd024cf8243

### CODE REVIEW

File: frontend/public/components/configmaps/ConfigMapFormEditor.tsx [hunk 1]
```diff
@@ -40,7 +40,8 @@ export const ConfigMapFormEditor: React.FC<FormikProps<any> & ConfigMapFormEdito
   const { t } = useTranslation();
   const { setFieldValue } = useFormikContext<ConfigMapFormInitialValues>();
   const LAST_VIEWED_EDITOR_TYPE_USERSETTING_KEY = 'console.configMapForm.editor.lastView';
-  const isStale = !!configMap && configMap?.metadata?.resourceVersion !== values.resourceVersion;
+  const isNew = !configMap?.metadata?.name || configMap?.metadata?.name === '~new';
```
[the remark this answers is not in the document]
Comment (rhamilto) [reply frontend/public/components/configmaps/ConfigMapFormEditor.tsx]: We've got that bogus logic [a few other places as well](https://github.com/search?q=repo%3Aopenshift%2Fconsole+%3F.metadata%3F.name+%3D%3D%3D+%27%7Enew%27&type=code).

I've updated this PR to the correct logic.

**Review by jhadvig — APPROVED** [review]
/lgtm

**Pull request merged**